--- a/src/core/exec-invoke.c
+++ b/src/core/exec-invoke.c
@@ -3268,8 +3268,10 @@ static int setup_exec_directory(
                                 goto fail;
                 }
 
-                /* Apply storage quotas and accounting */
-                r = apply_exec_quotas(target_dir, params->cgroup_path, type, &context->directories[type].exec_quota, &exec_dt_proj_id, &quota_already_enforced);
+                /* Apply storage quotas and accounting (skip in containers without cgroup) */
+                r = params->cgroup_path ?
+                        apply_exec_quotas(target_dir, params->cgroup_path, type, &context->directories[type].exec_quota, &exec_dt_proj_id, &quota_already_enforced) :
+                        0;
                 if (r < 0)
                         goto fail;
         }
